test/e2e/specs/editor/various/block-renaming.spec.js

```diff
@@ -145,6 +145,29 @@ test.describe( 'Block Renaming', () => {
 				},
 			] );
 		} );
+
+		test( 'does not allow renaming of blocks that do not support the feature', async ( {
+			editor,
+			page,
+			pageUtils,
+		} ) => {
+			// Only Group supports renaming.
+			await editor.insertBlock( {
+				name: 'core/paragraph',
+				attributes: { content: 'First Paragraph' },
+			} );
+
+			// Multiselect via keyboard.
+			await pageUtils.pressKeys( 'primary+a' );
+
+			// Expect the Rename control not to exist at all.
+			await expect(
+				page.getByRole( 'menuitem', {
+					name: 'Rename',
+					includeHidden: true, // the option is hidden behind modal
+				} )
+			).toBeHidden();
```

Comment:
Which part of the UI does this test? It's not clear from the test.

Reply:
The describe block provides the context. However the test itself could be made more focused by scoping the selectors to the relevant portion of the UI rather than the broad "rename" option.

Reply:
I've updated the scoping on the test. Feel free to merge if it looks good.